### ISSUE TITLE
cmake configure fails detection of libsafec /report of  SafeC feature option; uses incorrect var SAFEC_ENABLED

### ISSUE THREAD

**Issue by pgnd** [snort3/snort3#75]
with source from snort3/git
```
	cd snort3
	git log | head
		commit a8a93c26ae70fa5abca8daf547e41b229ad033fe
		Author: Michael Altizer (mialtize) <mialtize@cisco.com>
		Date:   Thu Jan 31 15:11:09 2019 -0500

		    Merge pull request #1483 in SNORT/snort3 from ~BBANTWAL/snort3:batching to master

		    Squashed commit of the following:

		    commit ecb607e1f70f760b545101a4dfa687f184aa2c36
		    Author: Jonathan McDowell <j.mcdowell@titan-ic.com>
```

and prereq safec libs built/installed
```
	cd safeclib
	git log | head
		commit 8f68fa3021a30f4bae081d6a158e5b8514579b10
		Author: Reini Urban <rurban@cpan.org>
		Date:   Sun Dec 30 11:16:50 2018 +0100

		    Release v30122018 3.4.0

		    See ChangeLog

		commit 7f1bfd5839dffbb801cd471ec99e79d6fb1d2c87
		Author: Reini Urban <rurban@cpan.org>

	find /usr/local/lib64 /usr/local/include -type f | grep safec
		/usr/local/lib64/libsafec-3.4.la
		/usr/local/lib64/libsafec-3.4.so.3.0.4
		/usr/local/lib64/pkgconfig/safec-3.4.pc
		/usr/local/include/libsafec/safe_lib_errno.h
		/usr/local/include/libsafec/safe_lib.h
		/usr/local/include/libsafec/safe_config.h
		/usr/local/include/libsafec/safe_str_lib.h
		/usr/local/include/libsafec/safe_mem_lib.h
		/usr/local/include/libsafec/safe_types.h
		/usr/local/include/libsafec/safe_compile.h

	pkg-config --libs --cflags safec-3.4
		-I/usr/local/include/safec-3.4 -I/usr/local/lib64/safec-3.4/include -L/usr/local/lib64 -lsafec-3.4
```

configuring snort3
```
	mkdir build
	cd    build
	cmake ..  \
	 -DENABLE_SAFEC=1 \
	 -DSAFEC_INCLUDE_DIR=/usr/local/include/libsafec \
	 -DSAFEC_LIBRARIES=/usr/local/lib64/libsafec-3.4.so \
	 ...
```

fails to find/use safec,
```
	Feature options:
	    Flatbuffers:    OFF
	    Hyperscan:      ON
	    ICONV:          ON
	    LZMA:           ON
	    RPC DB:         Built-in
	    SafeC:          OFF    <==================
	    TCMalloc:       OFF
	    UUID:           ON
```

Noting,
```
	grep SAFE CMakeCache.txt
!!!		ENABLE_SAFEC:BOOL=1
		SAFEC_INCLUDE_DIR:PATH=/usr/local/include/libsafec
		SAFEC_LIBRARIES:FILEPATH=/usr/local/lib64/libsafec-3.4.so
```

is inconsistent with CMakeLists.txt
```
	...
	if (HAVE_SAFEC AND SAFEC_ENABLED)
	    message("\
	    SafeC:          ON")
	else ()
	    message("\
	    SafeC:          OFF")
	endif ()
	...
```

SAFEC_ENABLED appears nowhere else in Snort3 source

also, in
```
	configure_cmake.sh (sh)  
        --disable-safec)
            append_cache_entry ENABLE_SAFEC             BOOL false
            ;;
        --enable-safec)
            append_cache_entry ENABLE_SAFEC             BOOL true
```

editing
```
	CMakeLists.txt
-		if (HAVE_SAFEC AND SAFEC_ENABLED)
+		if (HAVE_SAFEC AND ENABLE_SAFEC)
```

then re-
```
	cmake .. \
	...

		Feature options:
		    Flatbuffers:    OFF
		    Hyperscan:      ON
		    ICONV:          ON
		    LZMA:           ON
		    RPC DB:         Built-in
		    SafeC:          ON      <===========
		    TCMalloc:       OFF
		    UUID:           ON
```

What I'm not yet clear on is this *just* a Feature Option report bug?  Or if it goes deeper ...

**Comment by pgnd** [comment]
with that^ 'cheat' to get to an "ON" SafeC option, the subsequent build fails at memcpy_s ...
```
make V=1
	...
	[  3%] Building CXX object src/codecs/link/CMakeFiles/link_codecs.dir/cd_ciscometadata.cc.o
	[  3%] Building CXX object src/codecs/link/CMakeFiles/link_codecs.dir/cd_fabricpath.cc.o
	[  3%] Building CXX object src/codecs/link/CMakeFiles/link_codecs.dir/cd_mpls.cc.o
	/home/pgnd/dev/snort3/src/codecs/link/cd_mpls.cc: In member function ‘virtual bool {anonymous}::MplsCodec::encode(const uint8_t*, uint16_t, snort::EncState&, snort::Buffer&, snort::Flow*)’:
	/home/pgnd/dev/snort3/src/codecs/link/cd_mpls.cc:255:5: error: ‘memcpy_s’ was not declared in this scope
	     memcpy_s(buf.data(), hdr_len, hdr_start, hdr_len);
	     ^~~~~~~~
	/home/pgnd/dev/snort3/src/codecs/link/cd_mpls.cc:255:5: note: suggested alternative: ‘memcpy’
	     memcpy_s(buf.data(), hdr_len, hdr_start, hdr_len);
	     ^~~~~~~~
	     memcpy
	make[2]: *** [src/codecs/link/CMakeFiles/link_codecs.dir/build.make:128: src/codecs/link/CMakeFiles/link_codecs.dir/cd_mpls.cc.o] Error 1
	make[1]: *** [CMakeFiles/Makefile2:549: src/codecs/link/CMakeFiles/link_codecs.dir/all] Error 2
	make: *** [Makefile:152: all] Error 2
```

Don't know if it's related, or new ...

**Comment by Xiche** [comment]
Thanks for reporting this; let me get back to you on it.  We're in sort of an awkward position with SafeC in general because we built it against the much older version (from SourceForge) and Reini wasn't too interested in making things not tightly versioned (see my open PR and the associated Enhancement issue on his repository).  The miswording above for enabling is almost certainly a bug, but I'm not 100% where we want to go with SafeC integration at all right now.

**Comment by pgnd** [comment]
Given that uncertainty, what's the *current* recommendation for production?  Disable SafeC use?  TBH, I'm not entirely clear how integrated &/or necessary it is ...

**Comment by Xiche** [comment]
My current recommendation is to not use LibSafeC.

**Comment by pgnd** [comment]
@Xiche 

> My current recommendation is to not use LibSafeC.

What's, then, the correct way to (currently) DISABLE its use?

with
```
perl -pi -e 's|SAFEC_ENABLED|ENABLE_SAFEC|g' CMakeLists.txt
cmake .. -DENABLE_SAFEC=0 ...
```

make fails, still expecting libsafec  ...
```
[ 84%] Built target search_engines
[ 84%] Built target side_channel
[ 84%] Built target connectors
[ 85%] Built target file_connector
make[2]: *** No rule to make target '/usr/local/lib64/libsafec-3.4.so', needed by 'src/snort'.  Stop.
make[1]: *** [CMakeFiles/Makefile2:318: src/CMakeFiles/snort.dir/all] Error 2
make: *** [Makefile:152: all] Error 2
```

**Comment by Xiche** [comment]
By the way, this should be resolved in 1ffd89b3f6d0032f8ea1171b8c62ab967ae1223a.  If you use the latest safeclib from https://github.com/rurban/safeclib/ and have the pkgconfig dir it installed into in your PKG_CONFIG_PATH, it will try to find it and use it.  Thanks for reporting the enabling mixup.
